operator/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -982,6 +1002,59 @@ private boolean skipValidation(String mountPath) {
       return true;
     }
 
+    private void verifyCommonMounts() {
+      // if the common mount is specified, verify that specified volume exists in 'spec.commonMountVolumes'.
+      verifyCommonMounts(getAdminServerSpec().getCommonMounts());
+      getSpec().getManagedServers().forEach(managedServer -> verifyCommonMounts(managedServer.getCommonMounts()));
+    }
+
+    private void verifyCommonMounts(List<CommonMount> commonMounts) {
+      Optional.ofNullable(commonMounts)
+              .ifPresent(cmList -> cmList.forEach(this::checkIfVolumeExists));
+    }
+
+    private void checkIfVolumeExists(CommonMount cm) {
+      if (cm.getVolume() == null) {
+        failures.add(DomainValidationMessages.noCommonMountVolumeDefined());
+      } else if (Optional.ofNullable(getSpec().getCommonMountVolumes()).map(c -> c.stream()
+              .filter(commonMountVolume -> hasMatchingVolumeName(commonMountVolume, cm))
+              .collect(Collectors.toList())).orElse(new ArrayList<>()).isEmpty()) {
+        failures.add(DomainValidationMessages.noMatchingCommonMountVolumeDefined(cm.getVolume()));
+      }
+    }
+
+    private boolean hasMatchingVolumeName(CommonMountVolume commonMountVolume, CommonMount commonMount) {
+      return commonMount.getVolume().equals(commonMountVolume.getName());
+    }
+
+    private void verifyCommonMountVolumes() {
+      Optional.ofNullable(getSpec().getCommonMountVolumes())
+              .ifPresent(commonMountVolumes -> commonMountVolumes.forEach(this::checkNameAndMountPath));
+    }
+
+    private void checkNameAndMountPath(CommonMountVolume cmv) {
+      if (cmv.getName() == null) {
+        failures.add(DomainValidationMessages.commonMountVolumeNameNotDefined());
+      }
+    }
+
+    private void addDuplicateCommonMountVolumeNames() {
+      Optional.ofNullable(getSpec().getCommonMountVolumes())
+              .ifPresent(commonMountVolumes -> commonMountVolumes
+                      .forEach(this::checkDuplicateCommomMountVolume));
+    }
+
+    private void checkDuplicateCommomMountVolume(CommonMountVolume commonMountVolume) {
+      if (commonMountVolumes.stream().anyMatch(cmv -> cmv.getMountPath().equals(commonMountVolume.getMountPath()))) {
+        failures.add(DomainValidationMessages.duplicateCMVMountPath(commonMountVolume.getMountPath()));
+      } else if (commonMountVolumes.stream().anyMatch(cmv ->
+              cmv.getName().equals(toDns1123LegalName(commonMountVolume.getName())))) {
+        failures.add(DomainValidationMessages.duplicateCommonMountVolumeName(commonMountVolume.getName()));
+      } else {
+        commonMountVolumes.add(commonMountVolume);
```

Comment:
Minor: IMO, a check function should not have side effects unrelated to the check. This check adds cMV to commonMountVolumes...

Reply:
The `commonMountVolumes` HashSet here is only used for validation purposes (to check for a duplicate name or duplicate mount path). I followed the pattern similar to `checkDuplicateClusterName` method which does the following. I can change the name or create two HashSets (one for common volume name and one for mount path) if that makes it more clear.

    private void checkDuplicateClusterName(String clusterName) {
      if (clusterNames.contains(clusterName)) {
        failures.add(DomainValidationMessages.duplicateClusterName(clusterName));
      } else {
        clusterNames.add(clusterName);
      }
    }

Reply:
Oh - ok. Thank for explaining. Resolving.